immuneML/encodings/pdbencoding/PDBEncoder.py

```diff
@@ -0,0 +1,250 @@
+                                        list.append(atom.get_coord())
+
+                                    if "IMGT_CDR2" in region and (isInCDR2(atom.full_id[3][1]) or len(collection) >= 2):
+                                        list.append(atom.get_coord())
+
+                                    if "IMGT_CDR1" in region and (isInCDR1(atom.full_id[3][1]) or len(collection) >= 2):
+                                        list.append(atom.get_coord())
+
+
+                    saveList = list.copy()
+                    collection.append(saveList)
+                    counter = counter + 1
+                    list.clear()
+
+            lightID = -1
+            heavyID = -1
+            antigenID = -1
+
+            for i in range(0,len(collection)):
+                chainID = collection[i][0].id
+
+                if 'L' in chainID or 'A' in chainID:
+                    lightID = i
+
+                elif 'H' in chainID or 'B' in chainID:
+                    heavyID = i
+
+                elif 'P' in chainID or 'C' in chainID:
+                    antigenID = i
+
+                collection[i].pop(0)
+
+
+            if lightID >= 0 and heavyID >= 0 and antigenID == -1:
+                if lightID + heavyID == 1:
+                    antigenID = 2
+
+                elif lightID + heavyID == 3:
+                    antigenID = 0
+
+            elif lightID == -1  or heavyID == -1 or antigenID == -1:
+                lightID = 0
+                heavyID = 1
+                antigenID = 2
+
+
+            lightChainDistanceToAntigen = np.zeros(shape=(len(collection[antigenID]), len(collection[lightID])))
+
+            caluculateDistance(collection, lightID, antigenID, lightChainDistanceToAntigen)
+
+            heavyChainDistanceToAntigen = np.zeros(shape=(len(collection[antigenID]), len(collection[heavyID])))
+            caluculateDistance(collection, heavyID, antigenID, heavyChainDistanceToAntigen)
+
+            paddedArrays = paddingOfArray(lightChainDistanceToAntigen,heavyChainDistanceToAntigen)
+
+            structures.append([paddedArrays[0],paddedArrays[1]])
+            CDRcounter = CDRcounter +1
+
+
+        longestAntigen = 0
+        longestChain = 0
+
+        for x in structures:
+            if len(x[0]) > longestAntigen:
+                longestAntigen = len(x[0])
+            if len(x[0][0]) > longestChain:
+                longestChain = len(x[0][0])
+
+        paddedStructures = []
+        for x in structures:
+            paddedStructures.append(expandAntibodyChainToMaxLength(x, longestAntigen, longestChain))
+
+
+
+        numpArray = np.array(paddedStructures)
+
+        encoded_dataset = PDBDataset(self, dataset.pdbFilePaths,dataset.labels,dataset.metadata_file,EncodedData(numpArray,None,dataset.get_example_ids()))
+
+        #print("Shape of encoded_dataset.encoded_data.examples: " , numpy.shape(encoded_dataset.encoded_data.examples))
+        return encoded_dataset
+
+
```

Comment:
check the indentation; these are not methods, they are functions (i.e., they are placed outside the class! and no 'self') they should be methods instead though